test/unit/utils-test.coffee

```diff
@@ -0,0 +1,143 @@
+        # also add just one single line warning location
+        {index: str.indexOf('ten'), length: 3}
+      ]
+      ranges = utils.warningLocationToRanges location, str
+      assert.isArray ranges
+      assert.lengthOf ranges, 3
+      assert.deepEqual ranges, [
+        {start: 2, end: 4}
+        {start:6, end: 8}
+        {start: 10, end: 10}
+      ]
+
+    it 'works for some blueprint warnings too', (done) ->
+      blueprint = """
+      # Indented API
+
+      ## GET /url
+      + Response 200 (text/plain)
+
+        wrongly indented
+        resp.body
+
+      + Response 404 (text/plain)
+
+              ok indentation
+      """
+      protagonist.parse blueprint, (err, results) ->
+        return done err if err
+        assert.isObject results
+        assert.property results, 'warnings'
+        assert.isArray results.warnings
+        assert.lengthOf results.warnings, 1
+        assert.deepProperty results, 'warnings.0.location.0.index'
+        assert.deepProperty results, 'warnings.0.location.0.length'
+        ranges = utils.warningLocationToRanges results.warnings[0].location, blueprint
+        assert.isArray ranges
+        assert.lengthOf ranges, 1
+        assert.deepEqual ranges, [{start: 6, end: 7}]
+        done()
+
+    it 'returns an empty Array for empty locations', ->
+      assert.deepEqual utils.warningLocationToRanges([], placeholderText), []
+
+    it 'returns an empty Array for undefined locations', ->
+      assert.deepEqual utils.warningLocationToRanges(undefined, placeholderText), []
+
+  describe 'rangesToLinesText()', ->
+
+    describe 'when tested on fake locations', ->
+
+      it 'should return a string of line(s) separated with comma', ->
+        line = utils.rangesToLinesText [
+          {start: 2, end: 4}
+          {start: 8, end: 8}
+          {start: 10, end: 15}
+        ]
+        assert.strictEqual line, 'lines 2-4, line 8, lines 10-15'
+
+    describe 'for a real blueprint', ->
+      warnings = []
+      blueprint = null
+      ranges = []
+      before (done) ->
+        blueprint = """
+        # Indentation warnings API
+        ## GET /url
+        + Response 200 (text/plain)
+
+          badly indented 5. line
+          responsing body 6. line
+
+        + Response 400 (text/plain)
+
+          + Headers
+
+              headers-should-be:preformatted_12_line
+
+          + Body
+
+              not-enough indentation 16th line
+
+        ## POST /create
+        + Request (text/plain)
+
+            is it body?
+            maybe it is
+            if you say so
+
+        + Response 201
+
+                yup!
+        """
+        protagonist.parse blueprint, (err, results) ->
+          warnings = results.warnings or []
+          ranges = (utils.warningLocationToRanges(warn.location, blueprint) for warn in warnings)
+          done err
+
+      it 'shows ~ 4 warnings', ->
+        assert.lengthOf warnings, 4
+
+      it 'prints lines for those three warnings', ->
```

Comment:
Definitely four ;-)
Feel free to fix it.
Dne 18. 3. 2015 18:26 "Adam Kliment" <notifications@github.com> napsal(a):

> In test/unit/utils-test.coffee
> <https://github.com/apiaryio/dredd/pull/161#discussion_r26685990>:
>
> > +            maybe it is
> > +            if you say so
> > +
> > +        + Response 201
> > +
> > +                yup!
> > +        """
> > +        protagonist.parse blueprint, (err, results) ->
> > +          warnings = results.warnings or []
> > +          ranges = (utils.warningLocationToRanges(warn.location, blueprint) for warn in warnings)
> > +          done err
> > +
> > +      it 'shows ~ 4 warnings', ->
> > +        assert.lengthOf warnings, 4
> > +
> > +      it 'prints lines for those three warnings', ->
>
> three or four?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apiaryio/dredd/pull/161/files#r26685990>.
>